New in-wall engine sprites

AlinaStarkova opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
adds new variants for each sized engine to look like they are in walls (basically cut down sprites with their density set to false (because they dont take up the full tiles, especially the huge engines are 3x3 but only take up 1x3, they must have no density or they block way more than they should))

before: dumb looking gap
![image](https://user-images.githubusercontent.com/40489693/153381567-c968f58c-348b-4355-9812-55955b828011.png)



after: nice clean engine (and box trams are soul filled, also, trams kinda too small to really get any sort of diagnal design working)
![image](https://user-images.githubusercontent.com/40489693/153381679-bd87cfbf-9809-4f7e-899e-1b66e18ccab2.png)



showcase of each sized thruster:
![image](https://user-images.githubusercontent.com/40489693/153382365-cde9b73e-74ce-470d-827b-d27137bbe2a7.png)



added new structures that act like walls but arent turfs, and can only be used (and can be used!) on trams!
comes with 3 types, iron, plastitanium, and titanium. supports more types being added if you want but these are basically all that a tram needs to use:

https://user-images.githubusercontent.com/40489693/153407262-1dd2d5d2-c4d4-4538-a13c-e11ca18662d0.mp4



also removed a duplicated APC in a maint on tram, and removed cycle helpers from random maint airlocks that arent actually meant to cycle (they are solo)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

engines are kinda too bulky and organically shaped so they dont really fit some designs such as tram as a good example, in wall designs allow you to add thrusters onto any shaped ship easily without worrying about ensuring the walls around them being shaped right, as it appears to have the bulk of the engine merged inside the wall

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Nari Harimoto
imageadd: New thruster sprites to allow for in-wall designs
fix: removed cycle helpers from 2 random tram maint airlocks that arent meant to cycle (solo airlocks) and a duplicated APC in tram maint
expansion: the tram now has new walls! they are structures that function like walls so as to be built on the tram, create and deconstruct via the new tram girder (screwdriver on the girder to deconstruct, there is no displaced version ie wrench)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

AlinaStarkova committed:
in-wall engine sprites

AlinaStarkova committed:
no-diagnal

UnokiAs commented:
I will miss having my back burn from the engine in tram 😔 , Nice change 👍

AlinaStarkova committed:
tram walls!

AlinaStarkova commented:
initgration test has to do with mansus grab, not from this PR, and the mapdiffbot failed after 2s, so its a random failure, needs to be restarted

bruhlookatthisdood commented:
Maybe you could even add a little propulsion animation when it is activated?

AlinaStarkova commented:
> Maybe you could even add a little propulsion animation when it is activated?

the are currently just structures, fake blocks. engines dont actually do anything whatsoever. a moving shuttle can be engineless and work just fine. as such they have no code to support knowing if they are moving or not.

that said i could easily code it so if they are on the centcomm zlevel (when in hyperspace aka "moving" shuttles are actually on centcomm) they use the animations. however this will not work for tram, as the tram doesnt change z level, and im not sure if there is an easy way of detecting if its moving, and even then I'll have to detect what direction its moving, as if the tram moves left, the left side thrusters shouldnt activate

MrDoomBringer commented:
Or you could just make an animated sprite and click the 'movement state' button in dreammaker lol
But that's probably best saved for a different PR regardless. Good change 👍

MrDoomBringer commented:
> its the fact that there is thrusters facing two different directions on the tram

Yeah you're totally right - I hadn't thought this through. I had imagined having two separate states for left/right thrusters, but with how movement states work that wouldn't actually solve anything.
\</bikeshed>

Fikou commented on _maps/map_files/tramstation/tramstation.dmm:
```diff
@@ -20044,6 +20044,15 @@
 /area/maintenance/tram/mid)
+"fEi" = (
+/obj/structure/industrial_lift/tram{
+	icon_state = "invisible"
```
hm

AlinaStarkova replied:
yea i noticed that the tram floors aren't actually subtypes, they were just var edited to change their icon. and invisible is actually a floor icon. it allows me to place a tram floor under the thrusters without having a floor there so as to pull the thrusters along without having an ugly floor under it. its pretty neat and could probably become an actual subtype for the lift floors perhaps

Fikou commented on code/game/objects/structures/girders.dm:
```diff
@@ -171,6 +191,23 @@
+					to_chat(user, span_warning("You need at least two sheets to add plating!"))
+					return
+				to_chat(user, span_notice("You start adding plating..."))
+				if (do_after(user, 40, target = src))
```
use SECONDS defines for all these

AlinaStarkova replied:
to be fair, i basically copied false wall code since they act as walls while being structures, only i removed the false wall bits lol.

ill fix up the reviews when i get on my PC later

Fikou commented:
what youre doing here with turfs and girders seems incredibly cursed

AlinaStarkova committed:
uses SECONDS defines and

makes it so that welding off a tram wall takes as long as normal walls rather than instant

Maurukas reviewed (commented):
The map changes here seem to be incidental, so this needs to be reviewed from a code and sprite perspective. This seems to involve some duplication as well as touching the cursed realm of stack and sheet code.

optimumtact commented on code/game/objects/structures/girders.dm:
```diff
@@ -110,6 +116,20 @@
+						return
+					S.use(2)
+					to_chat(user, span_notice("You add the plating."))
+					var/obj/structure/tramwall/tram_wall = new S.tram_wall_type (loc)
```
please remove the space next to (loc)

AlinaStarkova replied:
done

AlinaStarkova committed:
Merge branch 'master' into new_engines

AlinaStarkova committed:
removes space

AlinaStarkova commented:
god dang merge conflicts left right and center

AlinaStarkova committed:
Merge branch 'master' into new_engines

optimumtact commented:
what are the changes to the map on tramstation

MrDoomBringer commented:
probably the new in-wall engines this pr adds

AlinaStarkova commented:
> what are the changes to the map on tramstation

literally shows in the OP, i changed the trams engine to the wall-engines, the entire purpose of the PR

AlinaStarkova committed:
Merge branch 'master' into new_engines

AlinaStarkova commented:
mapper and spriter approved this awhile ago, and this is the third time ive had to deal with a merge conflict :(

bruhlookatthisdood commented:
finnaly!
